client/src/pages/Forum/index.tsx

```diff
@@ -0,0 +1,80 @@
+import { useEffect, useState } from "react";
+import { Stack, Typography, useMediaQuery } from "@mui/material";
+import { Button } from "@mui/material";
+import { useGetForum } from 'api';
+import { ForumPostObject } from 'typings';
+import { useSearchParams } from 'react-router-dom';
+import SideDrawer from 'components/SideDrawer';
+import UserPost from 'components/UserPost';
+import Loading from 'components/Loading';
+
+export default function Forum() {
+    let startPosts: ForumPostObject[] = [];
+
+    const [searchParams, setSearchParams] = useSearchParams();
+    const [posts, setPosts] = useState(startPosts);
+    const [loading, setLoading] = useState(true);
+    const [refresh, setRefresh] = useState(true);
+    const [numPosts, setNumPosts] = useState(3);
```

Comment:
hard coded 3?

Reply:
So yeah in order to give it the number typing while still using the api call I gave it a default of 3, I changed this to be a '0' string now which defaults to getting all the posts if the num_posts parameter isn't specified. If we want something else i can change it